Minio/Helper/OperationsHelper.cs

```diff
@@ -15,11 +15,79 @@
  */
 
 using System;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Collections.Generic;
 using System.Linq;
 
+using Minio.DataModel;
+using System.IO;
+using RestSharp;
+
 namespace Minio
 {
+    public partial class MinioClient : IObjectOperations
+    {
+        /// <summary>
+        /// private helper method to remove list of objects from bucket
+        /// </summary>
+        /// <param name="args">GetObjectArgs Arguments Object encapsulates information like - bucket name, object name etc </param>
+        /// <param name="objectStat"> ObjectStat object encapsulates information like - object name, size, etag etc </param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
+        private async Task getObjectFileAsync(GetObjectArgs args, ObjectStat objectStat, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            long length = objectStat.Size;
+            string etag = objectStat.ETag;
+
+            long tempFileSize = 0;
+            string tempFileName = $"{args.FileName}.{etag}.part.minio";
+            if (!string.IsNullOrEmpty(args.VersionId))
+            {
+                tempFileName = $"{args.FileName}.{etag}.{args.VersionId}.part.minio";
+            }
+            if (File.Exists(args.FileName))
+            {
+                File.Delete(tempFileName);
+            }
+
+            utils.ValidateFile(tempFileName);
```

Comment:
what does this do?

Reply:
Validation that was previously there. It is to validate name before proceeding to save, so as to not encounter error during File operation.